### DISCUSSION TITLE
Allow question filtering by Eco-Score, Nutri-Score & more

### DISCUSSION THREAD

**Discussion by teolemon** [openfoodfacts/robotoff#1207] [Ideas]
### Problem
- Some apps or some completion pipeline might score-oriented, or device-oriented
- Some questions are thus irrelevant or unanswerable
- This becomes more complex if we start adding questions that require a photo (Can you take a photo of this ?)

### Proposed solution

- Allow question filtering by Eco-Score, Nutri-Score, NOVA/Ingredient analysis, and more

```
&mode=ecoscore (questions required to compute the Eco-Score)
&mode=nutriscore (questions required to compute Nutri-Score)
&mode=nova &mode=ingredients (questions required to compute NOVA)
&mode=yesno (yes/no questions only)
&mode=easy (easy questions)
&mode=nophoto (for questions that don't require a camera only)
&mode=photo (for photo questions only)
```

**Comment by alexgarel** [comment]
The modes could be stored in the json data of the insight
